Error when create or import cluster - panic: runtime error: invalid memory address or nil pointer dereference

Terraform version: 0.12.26  
Go runtime version: go1.12.13
found provider "terraform-provider-mongodbatlas_v0.6.0_x4"

```

Error: rpc error: code = Unavailable desc = transport is closing


panic: runtime error: invalid memory address or nil pointer dereference
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe0a598]
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4: 
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4: goroutine 41 [running]:
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4: github.com/terraform-providers/terraform-provider-mongodbatlas/mongodbatlas.flattenProcessArgs(...)
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-mongodbatlas/mongodbatlas/resource_mongodbatlas_cluster.go:1176
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4: github.com/terraform-providers/terraform-provider-mongodbatlas/mongodbatlas.resourceMongoDBAtlasClusterRead(0xc0002a8850, 0xfbd740, 0xc0004f4a00, 0xc0002a8850, 0x0)
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-mongodbatlas/mongodbatlas/resource_mongodbatlas_cluster.go:732 +0x2f38
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0000f4480, 0xc00008cf50, 0xfbd740, 0xc0004f4a00, 0xc00028ce10, 0x0, 0x0)
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-mongodbatlas/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:460 +0x119
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc0006da0d8, 0x135b560, 0xc000369a70, 0xc0006c2600, 0xc0006da0d8, 0xc000369a70, 0xc00054cb88)
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-mongodbatlas/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:525 +0x3d8
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler(0x1068e00, 0xc0006da0d8, 0x135b560, 0xc000369a70, 0xc0006c25a0, 0x0, 0x135b560, 0xc000369a70, 0xc0008c2480, 0x41d)
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-mongodbatlas/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3269 +0x217
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003f01a0, 0x1366a00, 0xc000202a80, 0xc000192300, 0xc000180f60, 0x1ac0eb0, 0x0, 0x0, 0x0)
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-mongodbatlas/vendor/google.golang.org/grpc/server.go:1082 +0x50a
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4: google.golang.org/grpc.(*Server).handleStream(0xc0003f01a0, 0x1366a00, 0xc000202a80, 0xc000192300, 0x0)
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-mongodbatlas/vendor/google.golang.org/grpc/server.go:1405 +0xcc9
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0002801d0, 0xc0003f01a0, 0x1366a00, 0xc000202a80, 0xc000192300)
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-mongodbatlas/vendor/google.golang.org/grpc/server.go:746 +0xa1
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-06-13T22:19:49.509-0300 [DEBUG] plugin.terraform-provider-mongodbatlas_v0.6.0_x4:  /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-mongodbatlas/vendor/google.golang.org/grpc/server.go:744 +0xa1
2020/06/13 22:19:49 [ERROR] <root>: eval: *terraform.EvalRefresh, err: rpc error: code = Unavailable desc = transport is closing
2020/06/13 22:19:49 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020-06-13T22:19:49.514-0300 [DEBUG] plugin: plugin process exited: path=/home/afonsorodrigues/projetos/meus-projetos/mongodb-terraform-example/.terraform/plugins/linux_amd64/terraform-provider-mongodbatlas_v0.6.0_x4 pid=19739 error="exit status 2"
2020/06/13 22:19:49 [TRACE] [walkImport] Exiting eval tree: import mongodbatlas_cluster.cluster_example result
2020/06/13 22:19:49 [TRACE] vertex "import mongodbatlas_cluster.cluster_example result": visit complete
2020/06/13 22:19:49 [TRACE] vertex "mongodbatlas_cluster.cluster_example (import id \"5ee571492eb6b363bdb4fe9b-clusterterraform\")": dynamic subgraph encountered errors
2020/06/13 22:19:49 [TRACE] vertex "mongodbatlas_cluster.cluster_example (import id \"5ee571492eb6b363bdb4fe9b-clusterterraform\")": visit complete
2020/06/13 22:19:49 [TRACE] dag/walk: upstream of "provider.mongodbatlas (close)" errored, so skipping
2020/06/13 22:19:49 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020/06/13 22:19:49 [TRACE] statemgr.Filesystem: removing lock metadata file .terraform.tfstate.lock.info
2020/06/13 22:19:49 [TRACE] statemgr.Filesystem: unlocking terraform.tfstate using fcntl flock
2020-06-13T22:19:49.515-0300 [DEBUG] plugin: plugin exited
```

When change version from "v0.6.0" to "v0.5.1", I created the cluster.

--- a/mongodbatlas/resource_mongodbatlas_cluster.go
+++ b/mongodbatlas/resource_mongodbatlas_cluster.go
@@ -1142,25 +1142,26 @@ func flattenRegionsConfig(regionsConfig map[string]matlas.RegionsConfig) []map[s
 
 func expandProcessArgs(d *schema.ResourceData, p map[string]interface{}) *matlas.ProcessArgs {
 	res := &matlas.ProcessArgs{}
-	if _, ok := d.GetOk("advanced_configuration.0.fail_index_key_too_long"); ok {
+
+	if _, ok := d.GetOkExists("advanced_configuration.0.fail_index_key_too_long"); ok {
 		res.FailIndexKeyTooLong = pointy.Bool(cast.ToBool(p["fail_index_key_too_long"]))
 	}
-	if _, ok := d.GetOk("advanced_configuration.0.javascript_enabled"); ok {
+	if _, ok := d.GetOkExists("advanced_configuration.0.javascript_enabled"); ok {
 		res.JavascriptEnabled = pointy.Bool(cast.ToBool(p["javascript_enabled"]))
 	}
-	if _, ok := d.GetOk("advanced_configuration.0.minimum_enabled_tls_protocol"); ok {
+	if _, ok := d.GetOkExists("advanced_configuration.0.minimum_enabled_tls_protocol"); ok {
 		res.MinimumEnabledTLSProtocol = cast.ToString(p["minimum_enabled_tls_protocol"])
 	}
-	if _, ok := d.GetOk("advanced_configuration.0.no_table_scan"); ok {
+	if _, ok := d.GetOkExists("advanced_configuration.0.no_table_scan"); ok {
 		res.NoTableScan = pointy.Bool(cast.ToBool(p["no_table_scan"]))
 	}
-	if _, ok := d.GetOk("advanced_configuration.0.sample_size_bi_connector"); ok {
+	if _, ok := d.GetOkExists("advanced_configuration.0.sample_size_bi_connector"); ok {
 		res.SampleSizeBIConnector = pointy.Int64(cast.ToInt64(p["sample_size_bi_connector"]))
 	}
-	if _, ok := d.GetOk("advanced_configuration.0.sample_refresh_interval_bi_connector"); ok {
+	if _, ok := d.GetOkExists("advanced_configuration.0.sample_refresh_interval_bi_connector"); ok {
 		res.SampleRefreshIntervalBIConnector = pointy.Int64(cast.ToInt64(p["sample_refresh_interval_bi_connector"]))
 	}
-	if _, ok := d.GetOk("advanced_configuration.0.oplog_size_mb"); ok {
+	if _, ok := d.GetOkExists("advanced_configuration.0.oplog_size_mb"); ok {
 		if sizeMB := cast.ToInt64(p["oplog_size_mb"]); sizeMB != 0 {
 			res.OplogSizeMB = pointy.Int64(cast.ToInt64(p["oplog_size_mb"]))
 		} else {
@@ -1173,10 +1174,10 @@ func expandProcessArgs(d *schema.ResourceData, p map[string]interface{}) *matlas
 func flattenProcessArgs(p *matlas.ProcessArgs) []interface{} {
 	return []interface{}{
 		map[string]interface{}{
-			"fail_index_key_too_long":              *p.FailIndexKeyTooLong,
-			"javascript_enabled":                   *p.JavascriptEnabled,
+			"fail_index_key_too_long":              cast.ToBool(p.FailIndexKeyTooLong),
+			"javascript_enabled":                   cast.ToBool(p.JavascriptEnabled),
 			"minimum_enabled_tls_protocol":         p.MinimumEnabledTLSProtocol,
-			"no_table_scan":                        *p.NoTableScan,
+			"no_table_scan":                        cast.ToBool(p.NoTableScan),
 			"oplog_size_mb":                        p.OplogSizeMB,
 			"sample_size_bi_connector":             p.SampleSizeBIConnector,
 			"sample_refresh_interval_bi_connector": p.SampleRefreshIntervalBIConnector,

--- a/mongodbatlas/resource_mongodbatlas_cluster_test.go
+++ b/mongodbatlas/resource_mongodbatlas_cluster_test.go
@@ -156,6 +156,53 @@ func TestAccResourceMongoDBAtlasCluster_basic_Partial_AdvancedConf(t *testing.T)
 	})
 }
 
+func TestAccResourceMongoDBAtlasCluster_emptyAdvancedConf(t *testing.T) {
+	var cluster matlas.Cluster
+
+	resourceName := "mongodbatlas_cluster.test"
+	projectID := os.Getenv("MONGODB_ATLAS_PROJECT_ID")
+	name := fmt.Sprintf("test-acc-%s", acctest.RandString(10))
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckMongoDBAtlasClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMongoDBAtlasClusterConfigAdvancedConfPartial(projectID, name, "false", &matlas.ProcessArgs{
+					MinimumEnabledTLSProtocol: "TLS1_2",
+				}),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMongoDBAtlasClusterExists(resourceName, &cluster),
+					testAccCheckMongoDBAtlasClusterAttributes(&cluster, name),
+				),
+			},
+			{
+				Config: testAccMongoDBAtlasClusterConfigAdvancedConf(projectID, name, "false", &matlas.ProcessArgs{
+					FailIndexKeyTooLong:              pointy.Bool(true),
+					JavascriptEnabled:                pointy.Bool(true),
+					MinimumEnabledTLSProtocol:        "TLS1_1",
+					NoTableScan:                      pointy.Bool(false),
+					OplogSizeMB:                      pointy.Int64(1000),
+					SampleRefreshIntervalBIConnector: pointy.Int64(310),
+					SampleSizeBIConnector:            pointy.Int64(110),
+				}),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMongoDBAtlasClusterExists(resourceName, &cluster),
+					testAccCheckMongoDBAtlasClusterAttributes(&cluster, name),
+					resource.TestCheckResourceAttr(resourceName, "advanced_configuration.0.fail_index_key_too_long", "true"),
+					resource.TestCheckResourceAttr(resourceName, "advanced_configuration.0.javascript_enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "advanced_configuration.0.minimum_enabled_tls_protocol", "TLS1_1"),
+					resource.TestCheckResourceAttr(resourceName, "advanced_configuration.0.no_table_scan", "false"),
+					resource.TestCheckResourceAttr(resourceName, "advanced_configuration.0.oplog_size_mb", "1000"),
+					resource.TestCheckResourceAttr(resourceName, "advanced_configuration.0.sample_refresh_interval_bi_connector", "310"),
+					resource.TestCheckResourceAttr(resourceName, "advanced_configuration.0.sample_size_bi_connector", "110"),
+				),
+			},
+		},
+	})
+}
+
 func TestAccResourceMongoDBAtlasCluster_basicAdvancedConf(t *testing.T) {
 	var cluster matlas.Cluster
 
@@ -870,14 +917,13 @@ func testAccMongoDBAtlasClusterConfigAdvancedConf(projectID, name, AutoscalingEn
 			name         = "%s"
 			disk_size_gb = 10
 			replication_factor           = 3
-			backup_enabled               = false 
+			backup_enabled               = false
 			auto_scaling_disk_gb_enabled =  %s
 			mongo_db_major_version       = "4.0"
 
 			// Provider Settings "block"
 			provider_name               = "AWS"
 			provider_disk_iops          = 100
-			provider_encrypt_ebs_volume = false
 			provider_instance_size_name = "M10"
 			provider_region_name        = "EU_CENTRAL_1"
 
@@ -902,14 +948,13 @@ func testAccMongoDBAtlasClusterConfigAdvancedConfPartial(projectID, name, Autosc
 			name         = "%s"
 			disk_size_gb = 10
 			replication_factor           = 3
-			backup_enabled               = false 
+			backup_enabled               = false
 			auto_scaling_disk_gb_enabled =  %s
 			mongo_db_major_version       = "4.0"
 
 			// Provider Settings "block"
 			provider_name               = "AWS"
 			provider_disk_iops          = 100
-			provider_encrypt_ebs_volume = false
 			provider_instance_size_name = "M10"
 			provider_region_name        = "EU_CENTRAL_1"
 

